app/src/main/java/org/fossasia/openevent/general/di/Modules.kt

```diff
@@ -212,7 +212,7 @@ val apiModule = module {
 
 val viewModelModule = module {
     viewModel { LoginViewModel(get(), get(), get()) }
-    viewModel { EventsViewModel(get(), get(), get(), get(), get(), get(), get()) }
+    viewModel { EventsViewModel(get(), get(), get(), get(), get(), get(), get(), get()) }
```

Comment:
This is getting too big. Open issue to refactor and break into smaller ViewModels

For e.g, similar event functionality should be in its own ViewModel

Reply:
Where is the issue?

Reply:
> Where is the issue?

https://github.com/fossasia/open-event-attendee-android/issues/2083

Reply:
> similar event functionality should be in its own ViewModel

It's not there, and there are only 4-5 methods, the view model is not so long. We can separate authentication related functions only.

Reply:
So why 8 parameters are required in it?

Reply:
```java
private val eventService: EventService,
    private val preference: Preference,
    private val resource: Resource,
    private val mutableConnectionLiveData: MutableConnectionLiveData,
    private val authHolder: AuthHolder,
    private val authService: AuthService,
    private val notificationService: NotificationService,
    private val config: PagedList.Config
```

Reply:
Can this be merged, issue #2083 is opened for it, will work on that.